types/unit/service.pp

```diff
@@ -39,5 +39,10 @@
     Optional['OOMScoreAdjust']            => Integer[-1000,1000],
     Optional['Environment']               => String,
     Optional['EnvironmentFile']           => Variant[Stdlib::Unixpath,Pattern[/-\/.*/]],
+    Optional['User']                      => String,
+    Optional['Group']                     => String,
+    Optional['StandardOutput']            => String,
+    Optional['StandardError']             => String,
+    Optional['WorkingDirectory']          => Stdlib::Unixpath,
```

Comment:
`Stdlib::Unixpath` actually looks too restrictive: https://www.freedesktop.org/software/systemd/man/systemd.exec.html#WorkingDirectory=

Also `~` and paths starting with `-` are allowed.